src/all/misskon/src/eu/kanade/tachiyomi/extension/all/misskon/MissKon.kt

```diff
@@ -0,0 +1,258 @@
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.concurrent.TimeUnit
+
+class MissKon() : SimpleParsedHttpSource() {
+
+    override val baseUrl = "https://misskon.com"
+    override val lang = "all"
+    override val name = "MissKon"
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient.newBuilder()
+        .rateLimitHost(baseUrl.toHttpUrl(), 10, 1, TimeUnit.SECONDS)
+        .setRandomUserAgent(UserAgentType.MOBILE)
+        .build()
+
+    override fun simpleMangaSelector() = "article.item-list"
+    override fun simpleMangaFromElement(element: Element): SManga {
+        val titleEL = element.select(".post-box-title")
+
+        val manga = SManga.create()
+        manga.title = titleEL.text()
+        manga.thumbnail_url = element.select(".post-thumbnail img").attr("data-src")
+        manga.setUrlWithoutDomain(titleEL.select("a").attr("href").substring(baseUrl.length))
+        return manga
+    }
+
+    override fun simpleNextPageSelector(): String? = null
+
+    // region popular
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/top3/")
+    // endregion
+
+    // region latest
+    override fun latestUpdatesRequest(page: Int): Request {
+        val url: String = if (page <= 1) {
+            baseUrl
+        } else {
+            "$baseUrl/page/$page"
+        }
+        return GET(url)
+    }
+
+    override fun latestUpdatesNextPageSelector() = ".current + a.page"
+    // endregion
+
+    // region Search
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val filter = filters.findInstance<TagFilter>()!!
+        return if (filter.isSelected()) {
+            GET(filter.selected.url)
+        } else {
+            GET("$baseUrl/page/$page/?s=$query")
+        }
+    }
+
+    override fun searchMangaNextPageSelector() = "div.content > div.pagination > span.current + a"
+    override fun searchMangaSelector() = "article.item-list"
+    // endregion
+
+    // region Details
+    override fun mangaDetailsParse(document: Document): SManga {
+        val postInnerEl = document.select("article > .post-inner")
+
+        val manga = SManga.create()
+        manga.title = postInnerEl.select(".post-title").text()
+        manga.description = ""
+        manga.genre = postInnerEl.select(".post-tag > a").joinToString(", ") { it.text() }
+        return manga
+    }
+
+    override fun chapterListSelector() = "html"
+
+    override fun chapterFromElement(element: Element): SChapter {
+        val dataSrc = element.selectFirst(".entry img")!!.attr("data-src")
```

Comment:
This is only used for the date, which is an optional field. Should not use non-null assert.